jumanji/environments/routing/rware/utils.py

```diff
@@ -0,0 +1,865 @@
+        grid: the warehouse floor grid array.
+        agents: a pytree of either Agent type containing agent information.
+        shelves: a pytree of Shelf type containing shelf information.
+        sensor_range: the range of the agent's sensors.
+        num_obs_features: the number of features in the observation array.
+        highways: binary array indicating highway positions.
+        agent_id: unique ID identifying a specific agent.
+
+    Returns:
+        a 1-d array containing the agent's observation.
+    """
+    agent = tree_slice(agents, agent_id)
+    agents_grid, shelves_grid = get_agent_view(grid, agent, sensor_range)
+
+    # write flattened observations
+    obs = jnp.zeros(num_obs_features, dtype=jnp.int32)
+    idx = 0
+
+    # write current agent position and whether carrying a shelf or not
+    obs, idx = write_to_observation(
+        obs,
+        idx,
+        jnp.array(
+            [agent.position.x, agent.position.y, agent.is_carrying],
+            dtype=jnp.int32,
+        ),
+    )
+
+    # write current agent direction
+    direction = jax.nn.one_hot(agent.direction, 4, dtype=jnp.int32)
+    obs, idx = write_to_observation(obs, idx, direction)
+
+    # write if agent is on highway or not
+    obs, idx = write_to_observation(
+        obs,
+        idx,
+        jnp.array(
+            [jnp.array(highways[agent.position.x, agent.position.y], int)],
+            dtype=jnp.int32,
+        ),
+    )
+
+    # function for writing receptive field cells
+    def write_no_agent(
+        obs: chex.Array, idx: int, _: int, is_self: bool
+    ) -> Tuple[chex.Array, int]:
+        "Write information for empty agent cell."
+        # if there is no agent we set a 0 and all zeros
+        # for the direction as well, i.e. [0, 0, 0, 0, 0]
+        idx = jax.lax.cond(is_self, lambda i: i, lambda i: move_writer_index(i, 5), idx)
+        return obs, idx
+
+    def write_agent(
+        obs: chex.Array, idx: int, id_agent: int, _: bool
+    ) -> Tuple[chex.Array, int]:
+        "Write information for cell containing an agent."
+        obs, idx = write_to_observation(obs, idx, jnp.array([1], dtype=jnp.int32))
+        direction = jax.nn.one_hot(
+            tree_slice(agents, id_agent - 1).direction, 4, dtype=jnp.int32
+        )
+        obs, idx = write_to_observation(obs, idx, direction)
+        return obs, idx
+
+    def write_no_shelf(obs: chex.Array, idx: int, _: int) -> Tuple[chex.Array, int]:
+        "write information for empty shelf cell."
+        idx = move_writer_index(idx, 2)
+        return obs, idx
+
+    def write_shelf(obs: chex.Array, idx: int, shelf_id: int) -> Tuple[chex.Array, int]:
+        "Write information for cell containing a shelf."
+        requested = tree_slice(shelves, shelf_id - 1).is_requested
+        obs, idx = write_to_observation(
+            obs,
+            idx,
+            jnp.array(
+                [1, requested],
+                dtype=jnp.int32,
+            ),
+        )
```

Comment:
Bit more clear :smile: 

```suggestion
        shelf = jnp.array([1, requested], dtype=jnp.int32)
        obs, idx = write_to_observation(obs, idx, shelf)
```